app/js/components/service_status.ts

```diff
@@ -0,0 +1,156 @@
+import * as Chart from 'chart.js';
+import {ChartTooltipItem} from "chart.js";
+
+interface StatesDictionary {
+  [key: string]: string;
+}
+
+interface EntityJson {
+  name: string,
+  environment: string,
+  link: string,
+}
+
+interface StatesResponseJson {
+  service: {
+    entities: EntityJson,
+    link: string,
+    name: string,
+    states: StatesDictionary,
+    labels: StatesDictionary,
+    tooltips: StatesDictionary,
+  }
+}
+
+class ServiceStatus {
+  private stateColors: StatesDictionary;
+  private context: CanvasRenderingContext2D;
+  private serviceId: string;
+  private chart: Chart|undefined;
+
+  constructor(context: CanvasRenderingContext2D, serviceId: string) {
+    this.context = context;
+    this.serviceId = serviceId;
+    this.stateColors = {};
+    this.chart = undefined;
+    this.initStates();
+  }
+
+  /**
+   * Init the eventhandlers on the elements
+   */
+  public registerEventHandlers() {
+    this.fetchState();
+  }
+
+  /**
+   * Check if chart is loaded
+   */
+  public isLoaded(): boolean
+  {
+    return this.chart != undefined;
+  }
+
+  private fetchState() {
+    const self = this;
+    $.ajax({
+      url: "/api/service/status/" + this.serviceId,
+      context: document.body
+    }).done(function (e) {
+      self.draw(<StatesResponseJson>e);
+    });
+  }
+
+  private draw(states: StatesResponseJson) {
+    let title = states.service.name;
+    let labels: string[] = [];
+    let tooltips: string[] = [];
+    let data: number[] = [];
+    let backgroundColor: string[] = [];
+
+    let ratio = 100 / Object.keys(states.service.states).length;
+    for (let key in states.service.states) {
+      const state = states.service.states[key];
+
+      data.push(ratio);
+      labels.push(states.service.labels[key]);
+      backgroundColor.push(this.stateColors[state]);
+      tooltips.push(states.service.tooltips[key])
+    }
+
+    let options: Chart.ChartConfiguration = {
+      type: 'doughnut',
+      data: {
+        labels: labels,
+        datasets: [{
+          data: data,
+          backgroundColor: backgroundColor
+        }]
+      },
+      options: {
+        cutoutPercentage: 75,
+        responsive: true,
+        legend: {
+          position: 'right',
+          labels: {
+            fontSize: 16
+          }
+        },
+        title: {
+          fontSize: 20,
+          display: true,
+          text: title,
+          position: 'left',
+        },
+        animation: {
+          animateScale: true,
+          animateRotate: true
+        },
+        tooltips: {
+          callbacks: {
+            label: function(tooltipItem: ChartTooltipItem) {
+              if (tooltipItem.index == undefined) {
+                return '';
+              }
+              return tooltips[tooltipItem.index];
```

Comment:
Is it possible to further style the tooltips to closer match the tooltips used throughout the project? For now, this already is super nice!

Reply:
Of course, but as I stated earlier I did't want to put too much effort in this